### PR TITLE
Fix tree collapse bug(fixes #293)

### LINKED ISSUE #293: Awkward collapse

### PR DESCRIPTION
**Pull request by sseliverstov**

### CODE REVIEW

File: allure-generator/src/main/javascript/components/tree/TreeView.js [hunk 1]
```diff
@@ -44,10 +39,10 @@ class TreeView extends View {
     onNodeClick(e) {
         this.$(e.currentTarget).parent().toggleClass('node__expanded');
         const uid = this.$(e.currentTarget).data('uid');
-        if (uid in  this.treeState) {
-            delete this.treeState[uid];
+        if (this.state.has(uid)) {
+            delete this.state.unset(uid, true);
```
Comment (just-boris) [inline allure-generator/src/main/javascript/components/tree/TreeView.js]: just do `this.state.unset(uid);`.
`delete` and `true` do nothing there.

Comment (sseliverstov) [inline allure-generator/src/main/javascript/components/tree/TreeView.js]: tnx

**Comment by sseliverstov** [comment]
@just-boris its ok now?

File: allure-generator/src/main/javascript/components/tree/TreeView.js [hunk 2]
```diff
@@ -44,10 +39,10 @@ class TreeView extends View {
     onNodeClick(e) {
         this.$(e.currentTarget).parent().toggleClass('node__expanded');
         const uid = this.$(e.currentTarget).data('uid');
-        if (uid in  this.treeState) {
-            delete this.treeState[uid];
+        if (this.state.has(uid)) {
+            this.state.unset(uid, true);
```
Comment (just-boris) [inline allure-generator/src/main/javascript/components/tree/TreeView.js]: `true` is not needed.
Look at the method signature: http://backbonejs.org/#Model-unset

**Pull request merged**